R/02-filter.R

```diff
@@ -75,14 +82,51 @@ gimap_filter <- function(.data = NULL,
   #then it finds the row sum (how many are filters flagged each construct e.g., number of TRUE in each row), 
   #and finally compares the row sum to the `min_n_filters` parameter to report TRUEs and FALSEs according to whether each construct is flagged by the minimum number of required filters
   #TRUE means it should be filtered, FALSE means it shouldn't be filtered
-  combined_filter <- rowSums(reduce(possible_filters, cbind)) >= min_n_filters
+  one_filter_df <- reduce(possible_filters, cbind) %>% 
+    `colnames<-`(c("FilterZeroCount", "FilterLowPlasmidCPM")) #*ADD any new filter's name here* as an additional column name; START with "Filter"
+  combined_filter <- rowSums(one_filter_df) >= min_n_filters
   #within `combined_filter` TRUE means that the filtering steps flagged the pgRNA construct for removal, therefore, we'll want to use the opposite FALSE values for the filtered data, keeping those that weren't flagged by filtering steps
   
-  gimap_dataset$filtered_data$filter_step_run <- TRUE #adding a way to know if the filter step was run since it's optional
+  
+  # store some data/results from filtering
+  ## adding a way to know if the filter step was run since it's optional
+  gimap_dataset$filtered_data$filter_step_run <- TRUE 
+  
+  ## subset the pgRNA IDs such that these are the ones that remain in the dataset following completion of filtering
   gimap_dataset$filtered_data$metadata_pg_ids <- gimap_dataset$metadata$pg_ids[!combined_filter,]
-  gimap_dataset$filtered_data$pg_metadata <- gimap_dataset$metadata$pg_metadata[!combined_filter,]
+  
+  ## subset the log2_cpm data such that these are the ones that remain in the dataset following completion of filtering
   gimap_dataset$filtered_data$transformed_log2_cpm <- gimap_dataset$transformed_data$log2_cpm[!combined_filter,]
-
+  
+  ## save a list of which pgRNAs are filtered out once filtering is complete
+  gimap_dataset$filtered_data$removed_pg_ids <- cbind(gimap_dataset$metadata$pg_ids[combined_filter,], one_filter_df[combined_filter,]) %>% #add the IDs as a column together with the TRUEs and FALSEs for each filter, focusing only on pgRNAs which are in some way flagged for removal
+                                                        pivot_longer(starts_with("Filter"), #pivot longer so that IDs are repeated and filter names are listed in a column and the last column (`boolVals`) are TRUEs and FALSEs
+                                                                     names_to = "filterName", 
+                                                                     values_to = "boolVals") %>% 
+                                                        filter(boolVals == TRUE) %>% #drop rows where boolVals is false, so this leaves only filters that flagged a pgRNA for removal
+                                                        select(id, filterName) %>% #drop the boolVals column because don't need it anymore
```

Comment:
```suggestion
                                                        select(id, filter_name) %>% #drop the boolVals column because don't need it anymore
```